### ISSUE TITLE
Environment not properly cleaned up after failed switch creation

### ISSUE THREAD

**Issue by AltGr** [ocaml/opam#4653]
If you try to create a switch and that fails, opam prompts you:
```
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] y
```

If you answer "y", then the switch is properly removed, but its entry remains in `PATH`. This is probably due to the fact that reverting the environment is based upon reading the `environment` file below the switch, and that no longer exists.

**Comment by kit-ty-kate** [comment]
I can't reproduce this. For instance with the following Dockerfile:
```
FROM ocaml/opam
RUN opam-2.1 switch create test --packages=conf-lua || true
RUN opam-2.1 env
```
The `test` switch doesn't remain in `PATH`

**Comment by AltGr** [comment]
Hm you're right something doesn't make sense in my report: I only had the shell environment in mind (not opam env files), but here even with shell_hooks setup, the new environment should never propagate to the parent shell.
Probably I answered `n` and removed `_opam` by hand at some point... in which case indeed the current approach won't work for cleanly restoring the env, but that's expected.

Closing for now, I'll see if I reproduce anything similar at some point.
